src/borg/archive.py

```diff
@@ -747,29 +747,60 @@ def compare_and_extract_chunks(self, item, fs_path, *, st, pi=None):
             # Second pass: Update file
             with backup_io("open"):
                 fs_file = open(fs_path, "rb+")
-            with fs_file:
```

Comment:
you removed the usage of the fs_file's context manager (removed the "with"). that means that it does not close the fs_file anymore (as it did when leaving the with-block), but I guess it should close the file.